extension/lib/shopgate/CartFactory.js

```diff
@@ -38,6 +39,8 @@ class ShopgateCartFactory {
       cart.addItem(cartItem)
     }
 
+    cart.addTotal('subTotal', 'SubTotal', subtotal)
```

Comment:
I've just noticed a flaw in previous design - previously we were just mapping so it was harder to see.
This calculation logic should be part of the Cart class not the factory. We can talk about how to do this.

Reply:
moved the logic